accessor_image_format/accessor_image_format.md

```diff
@@ -0,0 +1,166 @@
+| Original authors | Steffen Larsen steffen.larsen@codeplay.com, Alexander Johnston alexander@codeplay.com                                                                         |
+| Contributors     | Steffen Larsen steffen.larsen@codeplay.com, Alexander Johnston alexander@codeplay.com, Victor Lomüller victor@codeplay.com, Ruyman Reyes ruyman@codeplay.com  |
+
+# Image accessors with format information
+
+## Motivation
+
+In SYCL 1.2.1 operations on images are closely tied to the corresponding
+operations in OpenCL thus only allowing read and write operations to be done
+using 4-element vectors of 32-bit floating points, 32-bit signed integers,
+32-bit unsigned integers, or 16-bit floating points, which are in turn converted
+to the types of the channels in the image. Because of this, backends that cannot
+query the types of the channels in the image cannot determine the necessary
+conversions to do.
+
+## Overview
+
+This extension adds `sycl::short4`, `sycl::ushort4`, `sycl::schar4`, and
+`sycl::uchar4` to the list of types supported for `dataT` for image accessors.
+In order to carry information about the format of the image, the
+`image_format` enumerator from SYCL 2020 is adopted, given as
+
+```c++
+namespace sycl {
+enum class image_format : unsigned int {
+    r8g8b8a8_unorm,
+    r16g16b16a16_unorm,
+    r8g8b8a8_sint,
+    r16g16b16a16_sint,
+    r32b32g32a32_sint,
+    r8g8b8a8_uint,
+    r16g16b16a16_uint,
+    r32b32g32a32_uint,
+    r16b16g16a16_sfloat,
+    r32g32b32a32_sfloat,
+    b8g8r8a8_unorm,
+};
+} // namespace sycl
+```
+
+and mappings from `image_format` to the corresponding accessor data type is
+defined as
+
+```c++
+
+namespace sycl {
+template<image_format> struct image_format_type;
+
+template<> struct image_format_type<image_format::r8g8b8a8_unorm> {
+    using type = float4;
+};
+template<> struct image_format_type<image_format::r16g16b16a16_unorm> {
+    using type = float4;
+};
+template<> struct image_format_type<image_format::r8g8b8a8_sint> {
+    using type = schar4;
+};
+template<> struct image_format_type<image_format::r16g16b16a16_sint> {
+    using type = short4;
+};
+template<> struct image_format_type<image_format::r32b32g32a32_sint> {
+    using type = int4;
+};
+template<> struct image_format_type<image_format::r8g8b8a8_uint> {
+    using type = uchar4;
+};
+template<> struct image_format_type<image_format::r16g16b16a16_uint> {
+    using type = ushort4;
+};
+template<> struct image_format_type<image_format::r32b32g32a32_uint> {
+    using type = uint4;
+};
+template<> struct image_format_type<image_format::r16b16g16a16_sfloat> {
+    using type = half4;
+};
+template<> struct image_format_type<image_format::r32g32b32a32_sfloat> {
+    using type = float4;
+};
+template<> struct image_format_type<image_format::b8g8r8a8_unorm> {
+    using type = float4;
+};
```

Comment:
Could we maybe also define a helper alias:

```suggestion
};

template<image_format format>
using image_format_t = typename image_format_type<format>::type;
```

Reply:
Good call. I have added this and changed the data type of the accessor returned by the new `get_accessor` to use this.

I'm not completely convinced that `image_format_type` is the best name for this struct, so I'm open to alternatives.

Reply:
Perhaps `image_access_type`?

Reply:
I think that's a good name, though I'm not fond of the `_type` suffix. I've renamed it to `image_access`. What do you think?